[Gradle] Update nebula ospackage plugin to 12.1.0

breskeby opened the pull request:
- 12.1.0 is required by Gradle 9.x these days and required before
we can update to the newest Gradle version

breskeby committed:
[Gradle] Update nebula ospackage plugin to 12.1.0

- 12.1.0 is required by Gradle 9.x these days and required before
we can update to the newest Gradle version

breskeby committed:
Fix issue with file permissions in deb and rpm archives

jozala reviewed (approved):
Looks good.
One nit: It is worth to regenerate the `verification-metadata.xml` so that the old version entries got removed?

breskeby committed:
Cleanup dependency metadata

breskeby commented:
> Looks good. One nit: It is worth to regenerate the `verification-metadata.xml` so that the old version entries got removed?

I fixed the verification data and removed the outdated version.

 FYI if you run with `--write-verification-metadata sha256` it adds missing dependencies but does not delete any. Deleting the file and completely regenerating it is cumbersome as it contains some os specific dependencies, so you would not get that one right from scratch. So the only way to deal witih this is manually removing outdated versions.
